src/core/mesh/qgsmeshlayer.h

```diff
@@ -257,20 +257,22 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
 
     /**
       * Interpolates the value on the given point from given dataset.
+      * For edge, the returned value is on the projected position on the edge
```

Comment:
For 3D datasets, it uses \s dataset3dValue 
For 1D datasets, it uses \s dataset3dValue with `searchRadius`